packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.Props.ts

```diff
@@ -114,4 +116,167 @@ export interface ISpinButtonProps {
    * Icon for the decrement button of the spinButton
    */
   decrementButtonIcon?: IIconProps;
+
+  /**
+   * Custom styling for individual elements within the button DOM.
+   */
+  styles?: Partial<ISpinButtonStyles>;
+
+  /**
+   * Theme provided by HOC.
+   */
+  theme?: ITheme;
+}
+
+export interface ISpinButtonStyles {
+
+  /**
+   * Styles for the container of the spin button component.
+   */
+  container: IStyle;
+
+  /**
+   * Style for the label wrapper element of the component
+   * The label wrapper contains the icon and the label.
+   */
+  labelWrapper: IStyle;
+
+  /**
+   * Style override when the label is positioned at the start.
+   */
+  labelWrapperStart: IStyle;
+
+  /**
+   * Style override when the label is positioned at the end.
+   */
+  labelWrapperEnd: IStyle;
+
+  /**
+   * Style override when the label is positioned at the top.
+   */
+  labelWrapperTop: IStyle;
+
+  /**
+   * Style override when the label is positioned at the bottom.
+   */
+  labelWrapperBottom: IStyle;
+
+  /**
+   * Style for the icon.
+   */
+  icon: IStyle;
+
+  /**
+   * Style for the label text
+   */
+  label: IStyle;
+
+  /**
+   * Style for root when enabled.
+   */
+  root: IStyle;
+
+  /**
+   * Style override when
+   */
+  rootTopBottom: IStyle;
+
+  /**
+   * Style override when spinButton is enabled/hovered.
+   */
+  rootHovered: IStyle;
+
+  /**
+   * Style override when spinButton is enabled/focused.
+   */
+  rootFocused: IStyle;
+
+  /**
+   * Style override when spinButton is disabled.
+   */
+  rootDisabled: IStyle;
+
+  /**
+   * Styles for the input.
+   */
+  input: IStyle;
+
+  /**
+   * Style override for ::selection
+   */
+  inputTextSelected: IStyle;
+
+  /**
+   * Style override when spinButton is disabled.
+   */
+  inputDisabled: IStyle;
+
+  /**
+   * Styles for the arrowButtonsContainer
+   */
+  arrowButtonsContainer: IStyle;
+
+  /**
+   * Style override for the arrowButtonsContainer when spin button is disabled.
+   */
+  arrowButtonsContainerDisabled: IStyle;
+
+  /**
+   * Styles common for the upArrow and downArrow buttons.
+   */
+  arrowButtonStyles: Partial<ISpinButtonArrowButtonStyles>;
+
+  /**
+   * Style override for the upArrow button.
+   */
+  upArrowButtonStyles: Partial<ISpinButtonArrowButtonStyles>;
+
+  /**
+   * Style override for the downArrow button.
+   */
+  downArrowButtonStyles: Partial<ISpinButtonArrowButtonStyles>;
 }
+
+/**
+ * Styles for the arrow buttons.
+ *
+ * Note: The buttons are in a checked state when arrow keys are used to
+ * incremenent/decrement the spinButton. Use rootChecked instead of rootPressed
+ * for styling when that is the case.
+ */
+export interface ISpinButtonArrowButtonStyles extends IButtonStyles {
```

Comment:
I don't understand why you need an ISpinButtonArrowButtonStyles. Can't you just expose IButtonStyles directly?